pilot/pkg/serviceregistry/kube/controller.go

```diff
@@ -258,9 +257,7 @@ func (c *Controller) HasSynced() bool {
 // Run all controllers until a signal is received
 func (c *Controller) Run(stop <-chan struct{}) {
 	go func() {
-		if pilot.EnableWaitCacheSync {
```

Comment:
Why was this flag removed ? It was introduced for the same reasons the PR needs a flag, it had minimal testing ( 2-3 people with 'it starts' testing), on an area we had a lot of bugs: something goes wrong with one of the Apiservers in multicluster, and entire world goes down. Same problem when we switch to MCP. 

We can't just consider the simplest use case and do minimal testing few days before the release.

/hold

Reply:
>something goes wrong with one of the Apiservers in multicluster, and entire world goes down. 

This has nothing to do with waitcachesync. Even we donot wait sync, the watching for the apiserver is also down.